packages/flutter/lib/src/gestures/pointer_router.dart

```diff
@@ -94,19 +94,23 @@ class PointerRouter {
   /// Routes are called in the order in which they were added to the
   /// PointerRouter object.
   void route(PointerEvent event) {
-    final LinkedHashSet<_RouteEntry> routes = _routeMap[event.pointer];
-    final List<_RouteEntry> globalRoutes = List<_RouteEntry>.from(_globalRoutes);
+    final LinkedHashMap<PointerRoute, Matrix4> routes = _routeMap[event.pointer];
+    final LinkedHashMap<PointerRoute, Matrix4> copiedGlobalRoutes = LinkedHashMap<PointerRoute, Matrix4>.from(_globalRoutes);
     if (routes != null) {
-      for (_RouteEntry entry in List<_RouteEntry>.from(routes)) {
-        if (routes.any(_RouteEntry.isRoutePredicate(entry.route)))
-          _dispatch(event, entry);
-      }
-    }
-    for (_RouteEntry entry in globalRoutes) {
-      if (_globalRoutes.any(_RouteEntry.isRoutePredicate(entry.route)))
-        _dispatch(event, entry);
+      _dispatchEventToRoutes(event, routes, LinkedHashMap<PointerRoute, Matrix4>.from(routes));
     }
+    _dispatchEventToRoutes(event, _globalRoutes, copiedGlobalRoutes);
   }
+
+  void _dispatchEventToRoutes(
+    PointerEvent event,
+    LinkedHashMap<PointerRoute, Matrix4> referenceRoutes,
+    LinkedHashMap<PointerRoute, Matrix4> copiedRoutes,
+    ) => copiedRoutes.forEach((PointerRoute route, Matrix4 transform) {
```

Comment:
If you're concerned about performance in this code path: It may be faster to use a regular for loop over the keys then calling forEach. If you have a benchmark for this, you may want to try that.

Reply:
After reading this commit I thought that 'for-in' loop could be really faster, however, I decided to create benchmark tests for checking this supposition. And tests show that 'forEach' is faster than 'for-in'.

You can view these benchmark tests in the [repository](https://github.com/LehaIvanov/forin_vs_foreach_benchmark), two code blocks are compared there. These ones:
```
map.forEach((String key, int value) {
    ...
});
```
and 
```
for (String key in map.keys) {
    ...
}
```